src/internal/operators/switchScan.ts

```diff
@@ -0,0 +1,63 @@
+import { Observable } from '../Observable';
+import { ObservableInput, OperatorFunction, MonoTypeOperatorFunction } from '../types';
+import { Operator } from '../Operator';
+import { Subscriber } from '../Subscriber';
+import { of } from '../observable/of';
+import { switchMap } from './switchMap';
+import { tap } from './tap';
+
+export function switchScan<T>(accumulator: (acc: T, value: T, index: number) => ObservableInput<T>, seed?: T): MonoTypeOperatorFunction<T>;
+export function switchScan<T, R>(accumulator: (acc: R, value: T, index: number) => ObservableInput<R>, seed?: R): OperatorFunction<T, R>;
```

Comment:
I'm not sure why the seed is optional here, when it's not optional in `mergeScan`. Is there a particular reason for this? Also, the overload signatures here seem to have repeated a problem that the `scan` signatures have, IIRC.

If the accumulator is something other than `T`, a seed will be required, as without it, there will be no initial, non-`T` value that can be passed.

Reply:
I actually don't know what is the expected behavior here. I mentioned here https://github.com/ReactiveX/rxjs/issues/4348#issuecomment-450561459 that this is different even between `scan` and `mergeScan` so I made it closer to [`Array.reduce` that allows no seed value](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce#Syntax) as well.

Reply:
I understand now. If the `seed` param is mandatory then the `accumulator` function will always be called so this `T` and `R` mismatch won't happen.

Reply:
Yeah. There are a number of ways in which it could be resolved - including using `T | R` for the accumulator in the `R` signature without a seed and using `R` for the accumulator in a separate `R` signature with a seed. We should really have a look at the typings for `Array.prototype.reduce` that are within the TypeScript lib to see what's done there.

Reply:
https://github.com/Microsoft/TypeScript/blob/v3.3.1/lib/lib.es5.d.ts#L1168-L1180